ambry-cloud/src/main/java/com/github/ambry/cloud/BackupIntegrityMonitor.java

```diff
@@ -0,0 +1,132 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.cloud;
+
+import com.github.ambry.clustermap.CompositeClusterManager;
+import com.github.ambry.clustermap.DataNodeId;
+import com.github.ambry.clustermap.DiskId;
+import com.github.ambry.clustermap.PartitionId;
+import com.github.ambry.clustermap.ReplicaId;
+import com.github.ambry.config.ClusterMapConfig;
+import com.github.ambry.config.VerifiableProperties;
+import com.github.ambry.replication.FindTokenHelper;
+import com.github.ambry.replication.ReplicaThread;
+import com.github.ambry.replication.ReplicationManager;
+import com.github.ambry.store.StorageManager;
+import com.github.ambry.utils.Utils;
+import java.time.Duration;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static java.lang.Thread.*;
+
+
+/**
+ * Checks the integrity of backup partitions in Azure
+ */
+public class BackupIntegrityMonitor implements Callable<Integer> {
+  private final Logger logger = LoggerFactory.getLogger(RecoveryThread.class);
+  private final AtomicBoolean stop;
+  private final ClusterMapConfig clusterMapConfig;
+  private final CompositeClusterManager compositeClusterManager;
+  private final DataNodeId nodeId;
+  private final FindTokenHelper tokenFactory;
+  private final RecoveryManager azureReplicationManager;
+  private final RecoveryThread azureReplicator;
+  private final ReplicaThread serverReplicator;
+  private final ReplicationManager serverReplicationManager;
+  private final ScheduledExecutorService executor;
+  private final StorageManager storageManager;
+
+  public BackupIntegrityMonitor(RecoveryManager azure, ReplicationManager server,
+      CompositeClusterManager cluster, StorageManager storage, DataNodeId node, FindTokenHelper token,
+      VerifiableProperties properties) {
+    azureReplicationManager = azure;
+    azureReplicator = azure.getRecoveryThread("ambry_backup_integrity_monitor");
+    clusterMapConfig = new ClusterMapConfig(properties);
+    compositeClusterManager = cluster;
+    executor = Utils.newScheduler(1, "ambry_backup_integrity_monitor_", true);
+    nodeId = node;
+    serverReplicationManager = server;
+    serverReplicator = server.getBackupCheckerThread("ambry_backup_integrity_monitor");
+    stop = new AtomicBoolean(false);
+    storageManager = storage;
+    tokenFactory = token;
+    // log disk state
```

Comment:
@justinlin-linkedin added this, logging avlb disks on the node at startup, lmk if this is to be removed